backend/src/model/dto/timeline.rs

```diff
@@ -0,0 +1,34 @@
+use chrono::NaiveDate;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use typeshare::typeshare;
+use utoipa::{IntoParams, ToSchema};
+
+/// One summary datapoint in the timeline, used for a year, month or date
+#[typeshare]
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct TimelineEntryDto {
+    pub additions: i32,
+    pub removals: i32,
+}
+
+/// Used to summerize the data necessary for the timeline
+#[typeshare]
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct TimelineDto {
+    pub years: HashMap<String, TimelineEntryDto>,
+    pub months: HashMap<String, TimelineEntryDto>,
+    pub dates: HashMap<String, TimelineEntryDto>,
```

Comment:
The auto-generated example value of the endpoint is not great:

![image](https://github.com/ElektraInitiative/PermaplanT/assets/154217169/7e883c20-9d20-4008-9199-7cf28b999a02)


I suggest to manually write examples. Not sure if there is a better way to do it, but this works:
```suggestion
    #[schema(example = "{ \"2020\": { \"additions\": 7, \"removals\": 7 } }")]
    pub years: HashMap<String, TimelineEntryDto>,
    #[schema(example = "{ \"2020-01\": { \"additions\": 7, \"removals\": 7 } }")]
    pub months: HashMap<String, TimelineEntryDto>,
    #[schema(example = "{ \"2020-01-01\": { \"additions\": 3, \"removals\": 2 }, \"2020-01-05\": { \"additions\": 4, \"removals\": 5 } }")]
    pub dates: HashMap<String, TimelineEntryDto>,
```